### PR TITLE
Updated PR 44

### PR DESCRIPTION
**Pull request by nearmax**

### CODE REVIEW

File: borsh-rs/borsh/src/de/mod.rs [hunk 1]
```diff
@@ -196,7 +196,7 @@ where
             Err(io::Error::new(
                 io::ErrorKind::InvalidInput,
                 format!(
-                    "Invalid Result representation: {}. The first byte must be 0 or 1",
+                    "Invalid std::result::Result representation: {}. The first byte must be 0 or 1",
```
Comment (ilblackdragon) [inline borsh-rs/borsh/src/de/mod.rs:199]: too much replacing?

File: borsh-rs/borsh/src/de/mod.rs [hunk 2]
```diff
@@ -410,6 +410,12 @@ impl BorshDeserialize for Box<[u8]> {
     }
 }
 
+impl<T: BorshDeserialize> BorshDeserialize for Box<T> {
```
Comment (ilblackdragon) [inline borsh-rs/borsh/src/de/mod.rs:413]: hm, curious how Box<[u8]> above doesn't compete with this given now there is also [T] now. But if it works, that's fine.

File: borsh-rs/borsh/tests/test_de_errors.rs [hunk 3]
```diff
@@ -77,7 +77,7 @@ fn test_invalid_result() {
                 .unwrap_err()
                 .to_string(),
             format!(
-                "Invalid Result representation: {}. The first byte must be 0 or 1",
+                "Invalid std::result::Result representation: {}. The first byte must be 0 or 1",
```
Comment (ilblackdragon) [inline borsh-rs/borsh/tests/test_de_errors.rs:80]: Result is just fine.

File: borsh-rs/borsh/src/de/mod.rs (hunk 2 above)
Comment (nearmax) [reply borsh-rs/borsh/src/de/mod.rs:413]: Because `[T]` does not implement `BorshDeserialize` trait.

File: borsh-rs/borsh/src/de/mod.rs (hunk 1 above)
Comment (nearmax) [inline borsh-rs/borsh/src/de/mod.rs:199]:
```suggestion
                    "Invalid Result representation: {}. The first byte must be 0 or 1",
```

File: borsh-rs/borsh/tests/test_de_errors.rs (hunk 3 above)
Comment (nearmax) [inline borsh-rs/borsh/tests/test_de_errors.rs:80]:
```suggestion
                "Invalid Result representation: {}. The first byte must be 0 or 1",
```

**Pull request merged**